app/api/dds/v1/auth_providers_api.rb

```diff
@@ -28,6 +57,113 @@ class AuthProvidersAPI < Grape::API
       get '/auth_providers/:id', root: false, serializer: AuthenticationServiceSerializer do
         AuthenticationService.find(params[:id])
       end
+
+      desc 'List Auth Provider Affiliates' do
+        detail 'List Auth Provider Affiliates'
+        named 'list authentication provider affiliates'
+        failure [
+          [200, 'Success'],
+          [404, 'AuthenticationService does not exist'],
+          [400, 'AuthenticationService does not support affiliate search'],
+          [503, 'Identity Provider Communication Failure']
+        ]
+      end
+      params do
+        requires :id, type: String, desc: 'AuthenticationProvider UUID'
+        requires :full_name_contains, type: String, desc: 'string contained in name(must be at least 3 characters)'
+      end
+      get '/auth_providers/:id/affiliates', root: 'results', each_serializer: AffiliateSerializer do
+        affiliate_params = declared(params, {include_missing: false}, [:full_name_contains])
+        if UserAuthenticationService.where(
+          authentication_service_id: params[:id],
+          uid: affiliate_params[:uid]
+        ).exists?
+          affiliate_exists_error!
+        else
+          auth_service = AuthenticationService.find(params[:id])
+          if auth_service.identity_provider.nil?
+            unsupported_affiliate_search_error!
+          else
+            auth_service.identity_provider.affiliates(affiliate_params[:full_name_contains])
```

Comment:
I feel strongly that this endpoint needs to limit the number of returned results. The pagination default is 100, with a max of 1000. Ideally, the client would have the ability change this limit via the `per_page` parameter.

While it would be nice to do a full implementation of pagination, I don't believe that is necessary. That is, I don't think that we need to allow clients to use the `page` parameter. In fact, if `page` is provided, I suggest returning a "Not Implemented" or "Not supported by Identity Provider" error of some kind.

This is a technical limitation of ldap and that shouldn't dictate how our interface works. Let's say that you initially setup the affiliates endpoints with an identity provider that supported pagination and you were easily able to add support for `per_page` and `page`. Feature released, done. Now, you have to add a second identity provider that does not support pagination. What do you do? Remove pagination from the endpoint? Probably not. This is the way that I'm approaching this problem and hopefully provides context to my suggestions above.